### ISSUE TITLE
feat: implement different miners for hard and semi hard negative mining

### ISSUE THREAD

**Issue by gvondulong** [jina-ai/finetuner#233]
This feature will introduce different miners for hard and semi hard negative mining. 
Thereby, two miners (e.g. EasyHard) will be implemented for triplet learning. 
One miner will be implemented for siamese learning.

**Comment by gvondulong** [comment]
To validate that mining works, we compare fitting performance with and without mining activated. 
Learning rate is constant at 1e-5 and evaluation is performed each 5 epochs. 

The results below show

**Comment by tadejsv** [comment]
Great job @gvondulong !
